an-apluss/teamwork
# 28420421 Implement the Functionality for User(Admin/Employee) to Sign in

#### What does this PR do?
- Implement the functionality for user(admin/employee) to sign in

#### Description of Task to be completed?
- write the test suite for user endpoints
- seed dummy data into the database
- add the script to seed dummy data in the database
- implement the logic to log in a user
- add logic to compare if plaintext and hashed password is the same
- implement JWT to generate token for the user
- finish[ #6  ]

#### How should this be manually tested?
- Clone the repository using git clone `https://github.com/an-apluss/teamwork.git`  in your terminal
- run `npm install` on your terminal
- run `npm run migrations` on your terminal to create required tables for the application
- run `npm start` on your terminal to start the server
- make a `post` request on `https://web-teamwork.herokuapp.com/api/v1/auth/signin` with the necessary body request using postman
- use the below body request data
````
{
   "email": "anuoluwapoakinseye@gmail.com",
   "password": "secret"
}
````
- you should see response like below 
````
{
   "status": "success",
   "data": {
         "token": "sswwddwdww$djndjdj.cjdd.djd123._...",
         "userId": 1
   }
}
````
#### Any background context you want to provide?
Ensure that you provide value for the list of environment variable in the `.env.example`  file and rename to `.env`

#### What are the relevant issue stories?
- Github Project Board Card Link - ( [#28420421](https://github.com/an-apluss/teamwork/projects/1#card-28420421) )

#### Screenshots(if appropriate)
-N/A
